tests/components/telegram_bot/__init__.py

```diff
@@ -0,0 +1,7 @@
+"""Tests for telegram_bot component."""
```

Comment:
Add the requirements item from the telegram_bot integration manifest to the telegram manifest instead.

Reply:
Duplicate them you mean? Doesn't sound good maintenance wise. telegram just needs telegram_bot; what telegram_bot uses internally is its implementation detail.

Reply:
We rely on these requirements indirectly in the telegram integration. That's what the PR is fixing. Your proposal is another workaround to the flaw in our design. The correct fix is to merge these integrations but that's a breaking change.

If an integration requires test requirements it needs to have them in the manifest.

Reply:
Another way that would be ok is to move the imports in the telegram_bot integration inside functions that need them. This is how we do it for the mqtt integration.

Reply:
The import fiddling would be a workaround just like this one is, but a worse one IMO: we need tests fixed, but that would place the workaround in the actual homeassistant/ tree, away from what actually needs fixing, and in the tree that is more important to keep clean of workarounds and bad code in general than the tests/ tree.

In addition to the mentioned maintenance issue, the suggested dependency duplication in the manifest would be similarly placing the workaround further away from what needs fixing.

To me the approach in this PR is the least bad of the so far suggested evils.

Reply:
The right fix is that we make sure that we also add to `requirements_test_all.txt` all requirements of dependencies of integrations that have tests.

Reply:
Updated to do that, and updated `requirements_test_all.txt` accordingly as well. CI fails, maybe it's running the old gen_requirements_all and expecting its output, not one from the new?

Reply:
Run the script and it will update `requirements_test_all.txt`. If it doesn't, you didn't update the code correctly 👍

Reply:
I did run it, and it did update it :) And still does:
```
$ git diff -U0 origin/dev -- requirements_test_all.txt
# no changes, same as in dev
$ python -m script.gen_requirements_all
$ git diff -U0 origin/dev -- requirements_test_all.txt
diff --git a/requirements_test_all.txt b/requirements_test_all.txt
index 875cb1004c..7d83e5203a 100644
--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -25,0 +26,3 @@ PyRMVtransport==0.2.9
+# homeassistant.components.telegram_bot
+PySocks==1.7.1
+
@@ -855,0 +859,3 @@ python-tado==0.8.1
+# homeassistant.components.telegram_bot
+python-telegram-bot==11.1.0
+
```

Reply:
My local dev machine agrees with CI. See also my other comment.

Reply:
Do you mean that if you run the commands in the recipe as written above with this change applied, in your local setup requirements_test_all.txt does not change like shown above?

Reply:
Correct.